### ISSUE TITLE
integration with c++, golang, rust 

### ISSUE THREAD

**Issue by gengjiawen** [nodejs/single-executable#71]
Mainly inspired by this one https://github.com/patr0nus/libnode.

I think node.js can used as a powerful runtime embedded for other languages. Especially c++ which even doesn't have an http library or easy-to-use file operations.

Node.js have
* full v8 Engine
* full openssl
* extensive http support
* file operations
* other 

## What's lacking now

Official static lib. My prefer way would be a npm lib with all files，like https://www.npmjs.com/package/v8-android?activeTab=readme

**Comment by RaisinTen** [comment]
Sounds like this is more of an ask for the [embedder subsystem](https://nodejs.org/api/embedding.html) than for single-executables? In which case, should we move this to the core repo?

**Comment by mhdawson** [comment]
I agree both with embedding Node.js is interesting and that it likely fits better in the core repo as it's not related to SEA, at least in my current understanding.
